### PR TITLE
[DOC] Enhanced RDoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #===
- #to_s
- #inspect
- #strftme
- #asctime
- #iso3601
- #rfc3339
- #rfc2822
- #httpdate
- #jisx0301

### CODE REVIEW

File: ext/date/date_core.c [hunk 1]
```diff
@@ -6732,13 +6732,13 @@ cmp_dd(VALUE self, VALUE other)
  *
  *     d <=> DateTime.new(2022, 7, 26) # => 1
  *     d <=> DateTime.new(2022, 7, 27) # => 0
- *     d <=> DateTime.new(2022, 7, 29) # => -1
+ *     d <=> DateTime.new(2022, 7, 28) # => -1
  *
- * - A numeric (compares <tt>self.ajd</tt> to +other+):
+ * - A numeric (compares <tt>self.jd</tt> to +other+):
```
Comment (peterzhu2118) [inline ext/date/date_core.c:6737]: Why was this changed? It's compares `self.ajd`.

File: ext/date/date_core.c [hunk 2]
```diff
@@ -6804,20 +6804,39 @@ equal_gen(VALUE self, VALUE other)
...
- *    DateTime.new(2001,2,3) === DateTime.new(2001,2,3,12)
- *					#=> true
- *    DateTime.new(2001,2,3) === DateTime.new(2001,2,3,0,0,0,'+24:00')
- *					#=> true
- *    DateTime.new(2001,2,3) === DateTime.new(2001,2,4,0,0,0,'+24:00')
- *					#=> false
+ *   self === other -> true, false, or nil.
+ *
+ * Returns +true+ if +self+ and +other+ represent the same date,
+ * +false+ if not, +nil+ if the two are incommensurate.
```
Comment (peterzhu2118) [inline ext/date/date_core.c:6810]: Maybe "incommensurable"? At any rate, "incommensurate" is a very difficult word that even I (as a native speaker) didn't know and had to look it up in the dictionary. Maybe "comparable" is an easier word that means the same?

File: ext/date/date_core.c [hunk 3]
```diff
@@ -7192,13 +7212,15 @@ strftimev(const char *fmt, VALUE self,
 
 /*
  * call-seq:
- *    d.asctime  ->  string
- *    d.ctime    ->  string
+ *   asctime -> string
  *
- * Returns a string in asctime(3) format (but without "\n\0" at the
```
Comment (peterzhu2118) [inline ext/date/date_core.c:7198]: I think we should keep the fact that this is equivalent to the system `asctime` format.

File: ext/date/date_core.c [hunk 4]
```diff
@@ -7192,13 +7212,15 @@ strftimev(const char *fmt, VALUE self,
...
- * end).  This method is equivalent to strftime('%c').
+ * Equivalent to #strftime with argument <tt>'%a %b %e %T %Y'</tt>
+ * (or its {shorthand form}[https://docs.ruby-lang.org/en/master/strftime_formatting_rdoc.html#label-Shorthand+Conversion+Specifiers]
+ * <tt>'%c'</tt>);
  *
- * See also asctime(3) or ctime(3).
+ *   Date.today.asctime # => "Fri Jul 29 00:00:00 2022"
```
Comment (peterzhu2118) [inline ext/date/date_core.c:7221]: For all these examples, we should use `Date.new` to make it reproducible for the reader.

File: ext/date/date_core.c (hunk 1 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c]: Copy/edit mistake.

File: ext/date/date_core.c (hunk 2 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c]: Done (2 places).

File: ext/date/date_core.c (hunk 4 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c]: Done (8 places).

File: ext/date/date_core.c (hunk 3 above)
Comment (BurdetteLamar) [reply ext/date/date_core.c:7198]: Done (with link).

File: ext/date/date_core.c [hunk 5]
```diff
@@ -6732,13 +6732,13 @@ cmp_dd(VALUE self, VALUE other)
...
  *     d <=> DateTime.new(2022, 7, 27) # => 0
- *     d <=> DateTime.new(2022, 7, 29) # => -1
+ *     d <=> DateTime.new(2022, 7, 28) # => -1
  *
  * - A numeric (compares <tt>self.ajd</tt> to +other+):
  *
- *     d <=> 2459789 # => -1
  *     d <=> 2459788 # => -1
  *     d <=> 2459787 # => 1
+ *     d <=> 2459786 # => -1
```
Comment (peterzhu2118) [inline ext/date/date_core.c:6741]: I get `1` (rather than `-1`) returned for this example.

Comment (BurdetteLamar) [reply ext/date/date_core.c]: Indeed.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!
